Added new doc: WordPress Opentelemetry Instrumentation (#1418)
SigNoz/signoz.io: data/docs/instrumentation/opentelemetry-wordpress.mdx

---
date: 2025-05-05
id: opentelemetry-wordpress
title: WordPress Opentelemetry Instrumentation
description: Instrument WordPress with OpenTelemetry and send traces to SigNoz
tags: [opentelemetry, wordpress, instrumentation]
hide_table_of_contents: true
---

This document explains how to instrument an existing WordPress installation with OpenTelemetry and send traces to SigNoz using OpenTelemetry collector.

<figure data-zoomable align='center'>
    <img src="/img/docs/instrumentation/wordpress-otel-signoz.webp" alt="Image showing flow of traces"/>
    <figcaption><i>Send traces from WordPress to SigNoz using OpenTelemetry Collector</i></figcaption>
</figure>

> ## Requirements

Reviewer: ```suggestion
## Prerequisites
```

You will need the following in place before moving on to the next step:
- A self-managed WordPress site
- Permissions on the hosting server to install packages (Admin/Sudo)

> - [PHP 8.0+](https://www.php.net/) / PHP 8.1+ if you want to use _Protobuf_

Reviewer: ```suggestion
- [PHP 8.0+](https://www.php.net/) / PHP 8.1+ if you want to use [Protocols Buffers](https://protobuf.dev/) (_Protobuf_)

- [PECL](https://pecl.php.net/)
- [Composer](https://getcomposer.org/) 

<Admonition type="warning" title="Note">

> The instrumentation process involves performing tasks such as building the OpenTelemetry extension from source, modifying the `php.ini` file and restarting services.

Reviewer: ```suggestion
You'll need to perform tasks such as building the OpenTelemetry extension from source, modifying the `php.ini` file, and restarting services that may temporarily affect your website's availability.
```
@sangramrath can you confirm if the last part of this sentence is correct ? If not, skip adding " may temporarily affect your website's availability." this part.

Reviewer: Let's move this note to the setup section just below this "The following steps demonstrate how to instrument a WordPress site running on a Linux host with OpenTelemetry and then send the traces to SigNoz Cloud."

Author: @Calm-Rock yes restarting may temporarily affect services so we will keep it. Good catch!

</Admonition>

> ## Send WordPress traces to SigNoz Cloud

Reviewer: ```suggestion
## Setup

```bash
sudo pecl install opentelemetry
```

This will download the OpenTelemetry extension and its dependencies, compile them, and install them on the server in the default _php extensions_ folder. You may want to make a note of the path where the extensions were installed if needed for the `php.ini` configuration. 

Finally, you must enable the OpenTelemetry extension. You will not be able to install the _OpenTelemetry_ extensions using composer later if it is not enabled in PHP. To do so add the following line to your `php.ini` file:

```ini
[opentelemetry]
extension=opentelemetry.so
```

<Admonition type="info">
Ensure to modify the correct `php.ini` file that is being used by PHP.
</Admonition>

Finally, verify that the extensions are enabled by running the following command:

```bash
php -m | grep opentelemetry
```

If the extensions are installed correctly, you should see `opentelemetry` in the output.  

<Admonition type="info">
If `php -m` output is empty, try restarting the web server or if you are using PHP-FPM with NGINX, try restarting the `php-fpm` service.
</Admonition>

### Step 2: Add OpenTelemetry SDK and other dependencies

> In this step, you will use _Composer_ to add **OpenTelemetry SDK** and other dependencies such as the **OLTP Exporter** and the **OpenTelemetry auto-instrumentation for Wordpress** package for automatic instrumentation in WordPress. 

Reviewer: ```suggestion
In this step, you will use **Composer** to add **OpenTelemetry SDK** and other dependencies such as the **OTLP Exporter** and the **OpenTelemetry auto-instrumentation for Wordpress** package for automatic instrumentation in WordPress. 
```

> 

Reviewer: #### Navigate to root directory

Start by navigating to the root directory of your WordPress installation. This is usually the directory where `wp-config.php` is located.
For example, if your WordPress installation is located at `/var/www/html/wordpress`, run the following command:

```bash
cd /var/www/html/wordpress
```

> 

Reviewer: #### Initialise composer project

> Next, run the following command to initialize a new Composer project:

Reviewer: What if someone is already using composer ? Accomodate for that scenario as well.

```bash
composer init
```

> You will be prompted to enter some information about the project. You can skip most of the prompts by pressing `Enter` or `n`, but make sure to set the `name` and `description` fields. This will create a new `composer.json` file in the root directory of your WordPress installation.

Reviewer: ```suggestion
You will be prompted to enter some information about the project. You can skip most of the prompts but make sure to set the `name` and `description` fields. This will create a new `composer.json` file in the root directory of your WordPress installation.
```

> 

Reviewer: #### Add dependencies

Once the `composer.json` file is created, you will need to add the OpenTelemetry SDK and other dependencies. You can do this by running the following command:

```bash
composer require \
    open-telemetry/sdk \
    open-telemetry/exporter-otlp \ 
    open-telemetry/opentelemetry-auto-wordpress

> ```

Reviewer: Where to run this command ?

> Press `y` to any prompts during the installation process. The `composer.json` file should now look something like this:

Reviewer: >The `composer.json` file should now look something like this

This is with the assumption that there are no other extensions, i.e. a new project, which is generally not the case for production systems.

Author: Either ways the packages will be added to _require_. I have made sentence modifications that should do.

> ```json

Reviewer: ```suggestion
```json:composer.json
```

{

>     "name": "wordpress/opentelemtry-signoz",

Reviewer: ```suggestion
    "name": "wordpress/opentelemetry-signoz",
```

>     "name": "wordpress/opentelemtry-signoz",

Reviewer: Add a comment explaining what this name represent

Author: Added

    "require": {
        "open-telemetry/sdk": "^1.2",
        "open-telemetry/exporter-otlp": "^1.2",
        "symfony/http-client": "^6.4",
        "nyholm/psr7": "^1.8",
        "open-telemetry/opentelemetry-auto-wordpress": "^0.0.17"
    },
    "config": {
        "allow-plugins": {
            "php-http/discovery": true,
            "tbachert/spi": true
        }
    }
}

> ```

Reviewer: The versions should be replaced by placeholders since they will keep on changing with time.

Reviewer: Also, add a brief description about what these packages will do, For example:

This will add the core OpenTelemetry SDK for PHP, the OTLP exporter for sending data to SigNoz, HTTP client libraries for communication, and the [WordPress auto-instrumentation](https://github.com/opentelemetry-php/contrib-auto-wordpress) package to automatically capture WordPress operations.

Author: I have added a note stating that they may be different during the time of installation. 
And added the brief descriptions too.

### Step 3: Install the OpenTelemetry Collector

> In this step, you will install the OpenTelemetry collector on the WordPress host. The collector will receive traces from the OpenTelemetry PHP extension and send them to SigNoz. You will configure it to use the OTLP exporter for it.

Reviewer: ```suggestion
In this step, you will install the OpenTelemetry collector on the WordPress host. The collector acts as a middleware component that:
- Receives traces from your instrumented WordPress application
- Processes and batches the telemetry data efficiently
- Forwards the data to SigNoz using the OpenTelemetry Protocol (OTLP)
```

> To intall the OpenTelemetry collector, run the following command:

Reviewer: ```suggestion
To install the OpenTelemetry collector, run the following command:
```

```bash
wget https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.125.0/otelcol_0.125.0_linux_amd64.deb

> sudo dpkg -i otelcol_0.125.0_linux_amd64.deb

Reviewer: What if they are using the arm architecture for their Linux machine ?
Would it be better to hyperlink to [this doc](https://signoz.io/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/) ? Or maybe create two tabs ?

Author: I added a hyperlink to the release page and put a note. I hope that will do.

```

Verify that the collector is running with the following command:

```bash
sudo systemctl status otelcol
```

### Step 4: Configure PHP with OpenTelemetry settings

> In this step, you will configure PHP with OpenTelemtry settings. You will also add a prepend file to initialize composer. 

Reviewer: ```suggestion
In this step, you will configure PHP with OpenTelemetry settings. You will also add a prepend file to initialize composer. 
```

> Let's start with adding the following OpenTelemetry environment variables to the `php.ini` file. Choose an option based on your SigNoz setup i.e. whether you are using SigNoz Cloud or self-hosted SigNoz.

Reviewer: ```suggestion
Let's start with adding the following OpenTelemetry environment variables to the `php.ini` file. 
```

<Tabs entityName="plans">
<TabItem value="signoz-cloud" label="SigNoz Cloud" default>

Open the `php.ini` file in your favorite text editor and add the following lines:

```ini
OTEL_PHP_AUTOLOAD_ENABLED="true"

> OTEL_SERVICE_NAME=wordpress

Reviewer: ```suggestion
OTEL_SERVICE_NAME=<service-name>
```

OTEL_TRACES_EXPORTER=otlp
OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf

> OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4318

Reviewer: Where will the user use the ingestion enpoint for SigNoz Cloud ? For example -> `ingest.in.signoz.cloud`
Also, where will they use the ingestion key ?

Author: This has been fixed and added under 'OpenTelemetry Collector' section.

OTEL_PROPAGATORS=baggage,tracecontext
```

> <Admonition type="info">

Reviewer: ```suggestion
```

This should not be part of an admonition

A quick note on some of these values:

> - `wordpress` is the name of the service. This is the name that will be used to identify your service in SigNoz. Replace it with a different one as needed.

Reviewer: ```suggestion
- `<service-name>` is the name of your service, which will be used to identify your service in SigNoz.
```

- `http://localhost:4318` is the endpoint where OpenTelemetry collector is running, which in this example is the same host. Port 4318 is the default port for http protocol. 

> </Admonition>

Reviewer: ```suggestion
```

For more information on these environment variables, refer to the [OpenTelemetry PHP Configuration documentation](https://opentelemetry.io/docs/languages/php/sdk/#configuration).

</TabItem>

> <TabItem value="self-hosted-signoz" label="Self-Hosted SigNoz">

Reviewer: ```suggestion
<TabItem value="self-host" label="Self-Hosted SigNoz">
```

Open the `php.ini` file in your favorite text editor and add the following lines:

```ini
OTEL_PHP_AUTOLOAD_ENABLED="true"

> OTEL_SERVICE_NAME=wordpress

Reviewer: ```suggestion
OTEL_SERVICE_NAME=<service-name>
```

OTEL_TRACES_EXPORTER=otlp
OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
OTEL_EXPORTER_OTLP_ENDPOINT=http://self-host-ip
OTEL_PROPAGATORS=baggage,tracecontext
```

> <Admonition type="info">

Reviewer: ```suggestion
```

A quick note on some of these values:

> - `wordpress` is the name of the service. This is the name that will be used to identify your service in SigNoz. Replace it with a different one as needed.

Reviewer: ```suggestion
- `<service-name>` is the name of the service, which is used to identify your service in SigNoz. 
```

- `http://self-host-ip` is where the self-hosted SigNoz instance is running.

> </Admonition>

Reviewer: ```suggestion
```

For more information on these environment variables, refer to the [OpenTelemetry PHP Configuration documentation](https://opentelemetry.io/docs/languages/php/sdk/#configuration).

</TabItem>
</Tabs>

> 

Reviewer: #### Add a prepend file

> Next, to add a prepend file, you will use the `auto_prepend_file` directive in `php.ini` that allows you to specify a file to be executed before any other PHP files are run. This is useful for initializing the OpenTelemetry SDK and other dependencies. Modify the `auto_prepend_file` value to point to the `vendor/autoload.php` file that was created by Composer. Write the absolute path. 

Reviewer: ```suggestion
To add a prepend file, you will use the `auto_prepend_file` directive in `php.ini` that allows you to specify a file to be executed before any other PHP files are run. This is useful for initializing the OpenTelemetry SDK and other dependencies. 

Modify the `auto_prepend_file` value to point to the `vendor/autoload.php` file that was created by Composer. 

Write the absolute path. 
```

For example:

```ini
auto_prepend_file=/var/www/html/wordpress/vendor/autoload.php
```

> Finally, restart the web server or PHP-FPM service to apply the changes. 

Reviewer: ```suggestion
Now, restart the web server or PHP-FPM service to apply the configuration changes. 
```
